### PR TITLE
Json integration

### PR DESCRIPTION
**Pull request by jacquigiese**

### CODE REVIEW

File: CsvParser/csvParser.go [hunk 1]
```diff
@@ -85,7 +85,8 @@ func CsvParser(filePath string) [100]incident.DefaultIncident {
 	return incidentArray
 }
 
-func convertDefaultIncidentToIncident(arr []incident.DefaultIncident) []incident.Incident {
+//* This function takes in an array of 100 default incidents because that is what is provided by the csvParser above *//
+func convertDefaultIncidentToIncident(arr [100]incident.DefaultIncident) []incident.Incident {
```
Comment (amysorto) [inline CsvParser/csvParser.go]: Is there a benefit limiting to 100 incidents here versus when you call it (for example, in csvParser_test.go). Is there ever a case where we might need more than 100 incidents?

File: incident_viewer_demo/src/LineChart/LineChart.jsx [hunk 2]
```diff
@@ -178,8 +177,9 @@ function visProps(props) {
 class LineChart extends PureComponent {
   static propTypes = {
     annotationLineChunked: React.PropTypes.func,
-    goodIncidentSeries: React.PropTypes.object,
-    badIncidentSeries: React.PropTypes.object,
+    incident: React.PropTypes.object,
+    // goodIncidentSeries: React.PropTypes.object,
```
Comment (amysorto) [inline incident_viewer_demo/src/LineChart/LineChart.jsx]: remove these comments

File: incident_viewer_demo/src/LineChart/LineChart.jsx [hunk 3]
```diff
@@ -451,20 +450,6 @@ class LineChart extends PureComponent {
       .attr('class', 'highlight-ref-line');
     this.infoHoverBox = this.highlightDate.append('g')
 
-
-    // add in a rect to fill out the area beneath the hovered on X date
```
Comment (amysorto) [inline incident_viewer_demo/src/LineChart/LineChart.jsx]: Thanks for doing this!

File: incident_viewer_demo/src/LineChart/LineChart.jsx [hunk 4]
```diff
@@ -735,21 +719,21 @@ class LineChart extends PureComponent {
       // Take the series data, filter it for only in the good and bad ranges, then map those dates to their download speeds.
       // Use resulting download speeds to bound shaded regions.
       series[0].results
-        .filter(entry => (entry.date.isSameOrAfter(goodIncidentSeries.start.clone()) && entry.date.isSameOrBefore(goodIncidentSeries.end.clone())))
+        .filter(entry => (entry.date.isSameOrAfter(incident.goodPeriodStart.clone()) && entry.date.isSameOrBefore(incident.goodPeriodEnd.clone())))
```
Comment (amysorto) [inline incident_viewer_demo/src/LineChart/LineChart.jsx]: Can you remove the clone() part (and everywhere else it appears). This was leftover code for the shading that if added, will be added later.

File: CsvParser/csvParser.go (hunk 1 above)
Comment (jacquigiese) [reply CsvParser/csvParser.go]: This is a demo hack for this semester. We didn't want to want to process many more incidents than we thought we were going to use.

File: incident_viewer_demo/src/LineChart/LineChart.jsx [hunk 5]
```diff
@@ -280,10 +280,10 @@ class LineChart extends PureComponent {
    * @return {void}
    */
   onMouseMove(mouse) {
-    const { plotAreaHeight, goodIncidentSeries, badIncidentSeries, onHighlightDate, series, xScale , xKey, yScale, hasIncident } = this.props;
-    
-    const goodDescription1 = "Average Download Speed: 50 mb/s"
-    const goodDescription2 = "August 2015 - July 2016"
+    const { plotAreaHeight, incident, hasIncident, onHighlightDate, series, xScale , xKey, yScale, yKey } = this.props;
+    const goodDescription1 = incident.goodPeriodInfo
```
Comment (roman-rosenast) [inline incident_viewer_demo/src/LineChart/LineChart.jsx]: Could we get away without the '1' at the end of these var names?

File: incident_viewer_demo/src/LineChart/LineChart.jsx [hunk 6]
```diff
@@ -309,88 +309,87 @@ class LineChart extends PureComponent {
...
-            .attr('x', xScale(goodIncidentSeries.start) + 20)
+            .attr('x', xScale(incident.goodPeriodStart) + 20)
             .attr('dy', 0)
             .text(goodDescription1)
             .append('svg:tspan')
-            .attr('x', xScale(goodIncidentSeries.start) + 20)
+            .attr('x', xScale(incident.goodPeriodStart) + 20)
             .attr('dy', 20)
-            .text(goodDescription2)
+            // .text(goodDescription2)
```
Comment (roman-rosenast) [inline incident_viewer_demo/src/LineChart/LineChart.jsx]: I think we can remove this comment.

File: incident_viewer_demo/src/LineChart/LineChart.jsx [hunk 7]
```diff
@@ -309,88 +309,87 @@ class LineChart extends PureComponent {
...
+            .attr('x', xScale(incident.badPeriodStart) + 20)
             .attr('dy', 0)
-            .text(goodDescription1)
+            .text(badDescription1)
             .append('svg:tspan')
-            .attr('x', xScale(badIncidentSeries.start) + 20)
+            .attr('x', xScale(incident.badPeriodStart) + 20)
             .attr('dy', 20)
-            .text(goodDescription2)
+            // .text(goodDescription2)
```
Comment (roman-rosenast) [inline incident_viewer_demo/src/LineChart/LineChart.jsx]: Can we remove this comment?

File: incident_viewer_demo/src/LineChart/LineChart.jsx [hunk 8]
```diff
@@ -309,88 +309,87 @@ class LineChart extends PureComponent {
...
+        if (highlightedDate.isBefore(incident.goodPeriodEnd) && highlightedDate.isSameOrAfter(incident.goodPeriodStart) && mouseY > goodYmax) {
           this.infoHoverBox.append('rect')
           .classed("good-incident-area", true)
-          .attr('x', xScale(goodIncidentSeries.start))
+          .attr('x', xScale(incident.goodPeriodStart))
           .attr('y', goodYmax)
           .attr('width', width)
           .attr('height', plotAreaHeight-goodYmax);
 
           if (rectFitsText) {
```
Comment (roman-rosenast) [inline incident_viewer_demo/src/LineChart/LineChart.jsx]: One note about this block and the two similar ones below is that they have been changed in the textFix branch, which reformats the way that hover text is displayed. We should be deliberate about the fact that the textFix merge needs to use this JSON parsing and its own display formatting (ie: get rid of those tspans!).

File: incident_viewer_demo/src/LineChart/LineChart.jsx (hunk 2 above)
Comment (jacquigiese) [reply incident_viewer_demo/src/LineChart/LineChart.jsx]: Done

File: incident_viewer_demo/src/LineChart/LineChart.jsx (hunk 5 above)
Comment (jacquigiese) [reply incident_viewer_demo/src/LineChart/LineChart.jsx]: Done!

File: incident_viewer_demo/src/LineChart/LineChart.jsx (hunk 6 above)
Comment (jacquigiese) [reply incident_viewer_demo/src/LineChart/LineChart.jsx]: Done!

File: incident_viewer_demo/src/LineChart/LineChart.jsx (hunk 7 above)
Comment (jacquigiese) [reply incident_viewer_demo/src/LineChart/LineChart.jsx]: Done!

File: incident_viewer_demo/src/LineChart/LineChart.jsx (hunk 4 above)
Comment (jacquigiese) [reply incident_viewer_demo/src/LineChart/LineChart.jsx]: Done!

**Pull request merged**